### PR TITLE
Adjust iSCSI installation for SLE15

### PR DESCRIPTION
**Pull request by rwx788**
Test never worked for SLE15 and is still currently broken for all
distributions as requires changes in the backend
[PR#949](https://github.com/os-autoinst/os-autoinst/pull/949).

NOTE: see [PR#949](https://github.com/os-autoinst/os-autoinst/pull/949).

- [poo#23554](https://progress.opensuse.org/issues/23554)
- [Needles](https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/818)
- Verification runs:
  * [first_disk](http://g226.suse.de/tests/1732) refactoring didn't break it =D
  * [ibft SLE15](http://g226.suse.de/tests/1735)
  * [ibft SLE12](http://g226.suse.de/tests/1738)

### CODE REVIEW

File: tests/installation/partitioning_firstdisk.pm [hunk 1]
```diff
@@ -17,78 +17,9 @@ use warnings;
...
-    assert_screen 'use-entire-disk';
-    wait_screen_change {
-        send_key 'alt-e';
-    };
-    send_key $cmd{next};
-}
+use partition_setup 'take_first_disk';
 
 sub run {
-    if (is_storage_ng) {
```
Comment (asmorodskyi) [inline tests/installation/partitioning_firstdisk.pm]: such explicit call would make more sense as for me. There is no logic related to `take_first_disk_storage_ng` inside `take_first_disk` . You just hiding call to one function inside another while they fully independent.  
Before this change I could clearly see that we have two separate flows one for storage_ng and another for all other cases

Comment (rwx788) [reply tests/installation/partitioning_firstdisk.pm]: Yeah, and it was intended to abstract from storage stack, but focus on what we are doing. If you see take_first_disk, you know that first disk is selected irregardless of the flow. And if you want to know the details you open the method. What we can do is to check if it storage-ng or not when we call the method and then send it as parameter, but then we will do 2 checks for the same thing.

File: lib/partition_setup.pm [hunk 2]
```diff
@@ -218,5 +218,81 @@ sub enable_encryption_guided_setup {
...
+    }
+    assert_screen "btrfs-selected";
+    send_key $cmd{next};
+}
+
+sub take_first_disk {
+    my (%args) = @_;
+    if (is_storage_ng) {
+        take_first_disk_storage_ng %args;
+        return;
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]: Why not use here an `else`?
`return` in the middle of a function is the new `goto` ;)

I could agree with the use of a return when there is a significant performance impact. But here it isn't.

Comment (rwx788) [reply lib/partition_setup.pm]: I agree, that it will be more clear with else here, so fixed. Thanks!

**Pull request merged**